### ISSUE TITLE
Relevance Ranking in multilabel classification 

### ISSUE THREAD

**Issue by m-bittencourt** [scikit-multilearn/scikit-multilearn#133]
How to get a relevance ranking in scikit multilearn? I tried to use the estimates probability returned by predict_proba function. However, in ML-Knn for example, the most likely labels obtained in predict_proba aren't the same labels assigned by predict function.

**Comment by ChristianSch** [comment]
I am not entirely sure, but I don't think we have any real possibility to retrieve ranked labels. *However*, the commit message of https://github.com/scikit-multilearn/scikit-multilearn/commit/3ae1660e25438fc42ffec8fddc0073e584a547ca indicates that rankings can can be obtained via `predict_proba` from the model ML-kNN model. @niedakh noted that: " I am also adding a predict_proba function that assigns probabilities to labels instead of ranking them, as this is the general spirit of the scikit API. Such a vector can be easily ranked." (https://github.com/scikit-multilearn/scikit-multilearn/issues/22#issuecomment-278209682)

This sounds to me like the `predict_proba` method actually returns confidences in ranks. The `predict` method just seems to threshold them. I could be wrong though.

I am not really familiar with label ranking. If you could provide some basic code together with the expected output, I would look into it.

Relevant:
* https://github.com/scikit-multilearn/scikit-multilearn/issues/22#issue-92635759 an unverified method for `predict_ranking` is given by the issue op
* https://github.com/scikit-learn/scikit-learn/blob/55bf5d9/sklearn/metrics/ranking.py#L810 the code for `label_ranking_loss` in scikit-learn might be relevant as well

Maybe @niedakh can chime in as well, as he was the author of the commit back then.

**Comment by mbit9030** [comment]
Hi @ChristianSch,
Thanks for the answer and sorry for the delay in my return.
I need the labels ranking for to evaluate a new multi-label method and to compare with others. I want to use One-Error and Ranking-Loss ([here](https://github.com/scikit-learn/scikit-learn/blob/55bf5d9/sklearn/metrics/ranking.py#L810)) to measure the predict quality, evaluating the labels confidence.

I don't know if my idea is right or wrong, but I think that in the labels ranking, the labels that in are in the top are the most relevant. So, I would expected that _predict_ to return assigned the same top-ranking labels. However, in ML-KNN and in Binary Relevance methods, in ranking generated by _predict_proba_ probabilities, top-ranking labels not always are the same return in _predict_. I created a simple example: in sample test 0, for example, the top-ranking label isn't assigned in _predict_:

`from skmultilearn.dataset import load_dataset`
`from skmultilearn.adapt import MLkNN`

`D = load_dataset('enron', 'train')`
`X_train, y_train = D['X'], D['y']`

`D = load_dataset('enron', 'test')`
`X_test, y_test = D['X'], D['y']`
        
`classifier = MLkNN()`
`classifier.fit(X_train, y_train)`
`y_pred = classifier.predict(X_test)`
`vector_proba = classifier.predict_proba(X_test)`

Ranking labels, considering the _vector_proba_ probabilities of the first sample:

`ranking = sorted(range(vector_proba[0].shape[1]), key=lambda k: (vector_proba[0])[0,k], reverse=True)`
`ranking`
**`output -> [14, 25, 6, 11, 23, 5, 44, 13, 18, 46, 22, 9, 51, 39, ...]`**

The labels assined in _y_pred_ for the first sample:

`sp.sparse.find( y_pred[0] )[1]`
**`output -> array([ 6, 25], dtype=int32)`**


The top-ranking label (14) (the most likely) wasn't assigned in _y_pred_. I saw that a probability in _predict_proba_ is created based in a binary decision for each class, therefore the most likely labels aren't the same labels predicted. I think If I use _predict_proba_ for create a ranking, the One-Error or Ranking-Loss won't show the truth.

What do you think?

**Comment by northern-64bit** [comment]
I think that we can close this issue, since this got fixed in #193.
